tiledb/api/c_api/array_schema/array_schema_api_experimental.h

```diff
@@ -67,6 +67,36 @@ TILEDB_EXPORT capi_return_t tiledb_array_schema_timestamp_range(
     uint64_t* lo,
     uint64_t* hi) TILEDB_NOEXCEPT;
 
+/**
+ * Retrieves an already-loaded enumeration given its name.
+ *
+ * If the enumeration has already been loaded,
+ * such as via `tiledb_array_get_enumeration`,
+ * then this returns a pointer to the enumeration.
+ * Otherwise this returns `TILEDB_OK` and sets `enumeration` to NULL.
+ *
+ * **Example:**
+ *
+ * The following retrieves the enumeration named "states" in the schema.
+ *
+ * @code{.c}
+ * tiledb_enumeration_t* enmr;
+ * tiledb_array_schema_get_enumeration(ctx, array_schema, "states", &enmr);
+ * tiledb_attribute_free(&enmr);
+ * @endcode
+ *
+ * @param[in] ctx The TileDB context.
+ * @param[in] array_schema The array schema.
+ * @param[in] name The name of the enumeration to retrieve.
+ * @param[out] enmr The enumeration object to retrieve.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT capi_return_t tiledb_array_schema_get_enumeration(
```

Comment:
I think that the different behavior we are introducing here between `tiledb_array_get_enumeration` and `tiledb_array_schema_get_enumeration` APIs will be confusing to the user (as it was to me while reviewing) , ie that we expect form the former to load the enumeration if not loaded and from the latter not even though the syntax is so similar.
If we need one that doesn't load the enumeration I'd make this one explicit, eg. `tiledb_array_schema_get_loaded_enumeration` , and let `tiledb_array_schema_get_enumeration` (if added) have the same behavior as `tiledb_array_get_enumeration`.

Reply:
Took a peek and I see that we'd have to get a ClientResources something or other into ArraySchema (which @davisp did warn me about), doesn't seem worth it at the moment so we're going with the name change to `tiledb_array_schema_get_enumeration_if_loaded`.  0bf188d27